src/message.rs

```diff
@@ -295,6 +315,19 @@ impl Message {
     }
 }
 
+impl Default for Message {
+    fn default() -> Message {
+        Message {
+            subject: String::from(""),
+            reply: None,
+            data: Vec::new(),
+            headers: None,
+            client: None,
+            double_acked: Arc::new(AtomicBool::new(false)),
```

Comment:
Can remove all together imo.

Reply:
We want to drop

```
    /// Acknowledge a `JetStream` message and wait for acknowledgement from the server
    /// that it has received our ack. Retry acknowledgement until we receive a response.
    /// See `AckKind` documentation for details of what each variant means.
    ///
    /// Returns immediately if this message has already been double-acked.
```
    
behaviour?

Reply:
@derekcollison I can see that Go client has Ack behaviour with `ack uint`  that checks if message was already acked, but the behaviour differs.

Maybe we can look into parity regardless `ack` in separate issue?

Reply:
ack() should be fire and forget.
ack_sync() should wait for response, but not retry imo.

If Go client tracks double ack then we should here as well, did not realize it was doing that part.

Reply:
Ok, thats clear.

both Go acks does
```
	// Skip if already acked.
	if atomic.LoadUint32(&m.ackd) == 1 {
		return ErrInvalidJSAck
	}
```
before either sending publish (in case of `Ack()` which makes it fire and forget or  `Request` in case of `AckSync`.
Rust client has some difference in behaviour - acking acked message yelds `Ok` while Go `Err`.
Also Rust client retries in case of `double_ack` (which seems to aim the same behaviur as `AckSync()` in Go.

Reply:
ok so we should do similar in this client.